dts/bindings/i2c/st,stm32-i2c-v1.yaml

```diff
@@ -19,3 +19,17 @@ properties:
 
   pinctrl-names:
     required: true
+
+  scl-gpios:
```

Comment:
I wonder if there is a possibility to extend and use what is proposed [here](https://github.com/zephyrproject-rtos/zephyr/pull/54880/files#r1113056596) to retrieve directly gpio information from the pinctrl definition.
Do you need anything else than gpio port and pin ? (gpio flags ?)

Reply:
I agree, that's a much cleaner solution than requiring the gpio phandles. I'd prefer to wait until that's implemented somewhere generically accessible as is being discussed

Reply:
A remark on the macros introduced in my PR. Their purpose is to perform logical operations through DT based macros on phandle nodes within a pinctrl to provide insights on whether a particular function should be enabled.

What I am gathering is you would require to  know what phandle contains either the `scl` or `sda` function without caring that much about their actual pin and port designators?
 
Trying to get an accurate picture to assess how they could hopefully be useful for your case.

Reply:
I see, these use don't line up as well as I originally thought. Ultimately I'm looking to retrieve just enough information from a pinctrl definition to form a gpio spec for manual pin setting. For this I'm only interested in extracting the pin and port designators - flags should not matter.